checker/src/callbacks.rs

```diff
@@ -121,11 +122,12 @@ impl rustc_driver::Callbacks for MiraiCallbacks {
 }
 
 struct AnalysisInfo<'compilation, 'tcx> {
+    options: &'compilation Options,
     persistent_summary_cache: PersistentSummaryCache<'tcx>,
     constant_value_cache: ConstantValueCache<'tcx>,
     diagnostics_for: HashMap<DefId, Vec<DiagnosticBuilder<'compilation>>>,
     known_names_cache: KnownNamesCache,
-    analyze_single_func: Option<String>,
+    fn_whitelist: Option<Vec<String>>,
```

Comment:
rather spell out fn

Reply:
Why is this an Option?

Reply:
Settled offline for --diag=strict. Documented in the code why its an option, and renamed stuff.